NebulaHost/PacketProcessors/Factory/Entity/CreatePrebuildsRequestProcessor.cs

```diff
@@ -23,15 +23,19 @@ public void ProcessPacket(CreatePrebuildsRequest packet, NebulaConnection conn)
             }
 
             PlayerAction_Build pab = GameMain.mainPlayer.controller?.actionBuild;
-            if (pab != null)
+            BuildTool[] buildTools = GameMain.mainPlayer.controller?.actionBuild.tools;
+            BuildTool buildTool = null;
+            for (int i = 0; i < buildTools.Length; i++)
+            {
+                if(buildTools[i].GetType().ToString() == packet.BuildToolType)
```

Comment:
This is duplicated from a method in an earlier file - perhaps this should be made a utility method?

Reply:
Yes there is a lot of duplicate code within the client and host packet processors that I want to consolidate